bin/makePsfLibrary.py

```diff
@@ -0,0 +1,175 @@
+import lsst.afw.coord as afwCoord
+import lsst.afw.detection as afwDetection
+import lsst.afw.geom as afwGeom
+import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+import lsst.afw.table as afwTable
+import lsst.meas.base as measBase
+import lsst.meas.algorithms as measAlg
+
+psfSize = 33
+rafts = ["R01","R02","R03","R10","R11","R12","R13","R14","R20","R21","R22","R23","R24","R30","R31","R32","R33","R34","R41","R42","R43"]
+
+"""
+    Script to create a library of Psfs from an lsst focal plane.
+
+    @param[in]  catalog          A par file from PhoSim which lists all the stars
+    @param[in]  input_dir        Directory containing the fits.gz files from PhoSim
+    @param[in]  basename         prefix for the fits.gz files, excluding Rnn_Smm.fits.gz
+    @param[in]  output_dir       Directory in which to place the output libraries.
+
+    The output index is placed in output_dir/basename.index
+    The library is collected in fits files named output_dir/basename_Rnn.psfs.fits
+"""
+
+#   This routine takes a list of images and a catalog name (presumably covering the
+#   same area as the images.  Any entry in the catalog which is fully contained in an
+#   image is cutout, transformed to the center of the pupil, and then placed in a libray
+#   (multi-extension fits file).  The entry is also logged to an open file handle, fout
+def makeSubLibrary(images, catname, output_file, fout):
+    # Create an XYTransform from TanPixels to Pixels.  This is the transform
+    # needed to remove the optical distortion from the Psf. Use R22_S11 as center
+    butler = dafPersist.Butler(root=os.path.join(os.path.dirname(__file__), "."))
+    camera = butler.get("camera", immediate=True)
+    det = camera["R:2,2 S:1,1"]
+    tm = det.getTransformMap()
+    for sys in tm.getCoordSysList():
+        if sys.getSysName() == "Pixels":
+            transform1 = tm.get(sys).invert()
+        if sys.getSysName() == "TanPixels":
+            transform2 = tm.get(sys)
+    xytransform = afwGeom.MultiXYTransform([transform1, transform2])
+    
+    # A centroid algorithm has to be run here, because the catalog from OpSim
+    # is not very accurate
+    schema = afwTable.SourceTable.makeMinimalSchema()
+    control = measBase.GaussianCentroidControl()
+    alg = measBase.GaussianCentroidAlgorithm
+    plugin = alg(control, "", schema)
+    
+    # count of psfs actually created
+    psfIndex = 1
+    # Go through all the images and write out the galaxies
+    psfCount = 0
+    outputHdus = pyfits.hdu.hdulist.HDUList()
+    for image in images:
+        exp = afwImage.ExposureF(image)
+        cat = open(catname, "r")
+        # number the psfs the same way they are listed in the catalog, starting with line 1
+        # read through the OpSim catalog and create a psf for any object which is completely
+        # inside the image we are processing
+        while True:
+            line = cat.readline()
+            if len(line) == 0:
+                break
```

Comment:
I guess the aim here is to iterate through each line in ``catname``. I'd suggest using a context manager to open the file (that way, you don't have to remember to close it), and a more idiomatic loop. Something like this:

```python
with open(catname, "r") as cat:
    for line in cat:
        # do stuff
```